src/libImaging/TiffDecode.c

```diff
@@ -558,65 +589,89 @@ ImagingLibTiffDecode(
-                     * backwards
-                     */
-
+            UINT8 plane;
+            for (plane = 0; plane < planes; plane++) {
+                ImagingShuffler shuffler = unpackers[plane];
+                for (x = state->xoff; x < state->xsize; x += tile_width) {
                     if (isYCbCr) {
-                        current_line = tile_length - tile_y - 1;
+                        /* To avoid dealing with YCbCr subsampling, let libtiff handle it */
+                        if (!TIFFReadRGBATile(tiff, x, y, (UINT32 *)state->buffer)) {
+                            TRACE(("Decode Error, Tile at %dx%d\n", x, y));
+                            state->errcode = IMAGING_CODEC_BROKEN;
+                            goto decode_err;
+                        }
                     } else {
-                        current_line = tile_y;
+                        if (TIFFReadTile(tiff, (tdata_t)state->buffer, x, y, 0, plane) == -1) {
+                            TRACE(("Decode Error, Tile at %dx%d\n", x, y));
+                            state->errcode = IMAGING_CODEC_BROKEN;
+                            goto decode_err;
+                        }
                     }
 
-                    state->shuffle(
-                        (UINT8 *)im->image[tile_y + y] + x * im->pixelsize,
-                        state->buffer + current_line * row_byte_size,
-                        current_tile_width);
+                    TRACE(("Read tile at %dx%d; \n\n", x, y));
+
+                    current_tile_width = min((INT32) tile_width, state->xsize - x);
+                    current_tile_length =  min((INT32) tile_length, state->ysize - y);
+                    // iterate over each line in the tile and stuff data into image
+                    for (tile_y = 0; tile_y < current_tile_length; tile_y++) {
+                        TRACE(("Writing tile data at %dx%d using tile_width: %d; \n", tile_y + y, x, current_tile_width));
+
+                        // UINT8 * bbb = state->buffer + tile_y * row_byte_size;
+                        // TRACE(("chars: %x%x%x%x\n", ((UINT8 *)bbb)[0], ((UINT8 *)bbb)[1], ((UINT8 *)bbb)[2], ((UINT8 *)bbb)[3]));
+                        /*
+                         * For some reason the TIFFReadRGBATile() function
+                         * chooses the lower left corner as the origin.
+                         * Vertically mirror by shuffling the scanlines
+                         * backwards
+                         */
+
+                        if (isYCbCr) {
+                            current_line = tile_length - tile_y - 1;
+                        } else {
+                            current_line = tile_y;
+                        }
+
+                        shuffler((UINT8*) im->image[tile_y + y] + x * im->pixelsize,
+                                 state->buffer + current_line * row_byte_size,
+                                 current_tile_width
+                                 );
```

Comment:
The error doesn't show up when compiling with in a debug build (`setup.py build_ext --dubug`), I had to change `setup.py` line 852 to `Extension("PIL._imaging", files, extra_compile_args=["/Z7"], extra_link_args=["/DEBUG"]),` and copy the PDB to the PIL directory.

I'm not yet sure what is going on, but compiling with debug information shows that `state->buffer` is NULL on this line even though it was valid on line 587 for the realloc call.

Reply:
Solved it, see https://github.com/wiredfool/Pillow/pull/6